common/flatpak-context.c

```diff
@@ -614,69 +615,71 @@ get_xdg_user_dir_from_string (const char  *filesystem,
       if (config_key)
         *config_key = "XDG_DESKTOP_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_DESKTOP);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_DESKTOP));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-documents") == 0)
     {
       if (config_key)
         *config_key = "XDG_DOCUMENTS_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_DOCUMENTS);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_DOCUMENTS));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-download") == 0)
     {
       if (config_key)
         *config_key = "XDG_DOWNLOAD_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_DOWNLOAD);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_DOWNLOAD));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-music") == 0)
     {
       if (config_key)
         *config_key = "XDG_MUSIC_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_MUSIC);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_MUSIC));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-pictures") == 0)
     {
       if (config_key)
         *config_key = "XDG_PICTURES_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_PICTURES);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_PICTURES));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-public-share") == 0)
     {
       if (config_key)
         *config_key = "XDG_PUBLICSHARE_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_PUBLIC_SHARE);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_PUBLIC_SHARE));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-templates") == 0)
     {
       if (config_key)
         *config_key = "XDG_TEMPLATES_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_TEMPLATES);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_TEMPLATES));
       return TRUE;
     }
   if (strcmp (prefix, "xdg-videos") == 0)
     {
       if (config_key)
         *config_key = "XDG_VIDEOS_DIR";
       if (dir)
-        *dir = g_get_user_special_dir (G_USER_DIRECTORY_VIDEOS);
+        *dir = g_strdup (g_get_user_special_dir (G_USER_DIRECTORY_VIDEOS));
       return TRUE;
     }
-  if (get_xdg_dir_from_prefix (prefix, NULL, dir))
+  if (get_xdg_dir_from_prefix (prefix, NULL, &dir2))
```

Comment:
`dir2` is not a good name. Maybe call this `xdg_dir` or rename the out variable to `dir_out`?

Reply:
`dir_out`.